OData/src/System.Web.OData/OData/ODataSwaggerConverter.cs

```diff
@@ -0,0 +1,287 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Web.Http;
+using Microsoft.OData.Edm;
+using Newtonsoft.Json.Linq;
+
+namespace System.Web.OData
+{
+    /// <summary>
+    /// Represents an <see cref="ODataSwaggerConverter"/> used to converter an Edm model to Swagger model.
+    /// </summary>
+    public class ODataSwaggerConverter
+    {
+        private static readonly Uri DefaultMetadataUri = new Uri("http://localhost");
+        private const string DefaultHost = "default";
+        private const string DefaultbasePath = "default";
+
+        private JObject _swaggerDoc;
+        private JObject _swaggerPaths;
+        private JObject _swaggeDefinitions;
```

Comment:
Can we make these as protected, so customer can change them by overriding particular function?